### DISCUSSION TITLE
Json parsing error despite being accepted

### DISCUSSION THREAD

**Discussion by MikailKahya** [nlohmann/json#3990] [Q&A] [answered]
I have a basic json file which gets checked whether it is possible. According to the function it is a valid json file but the gives me a parsing error when exceptions are allowed and discards it when it is not. Anyone see what I am doing wrong as it is a valid json file and loads in fine

<img width="573" alt="image" src="https://user-images.githubusercontent.com/91017318/226845622-318085f9-7e78-4cea-b30f-b94880c78175.png">
<img width="417" alt="image" src="https://user-images.githubusercontent.com/91017318/226845797-01752ca0-7caa-4981-b60e-7eb800db0ab6.png">
<img width="200" alt="image" src="https://user-images.githubusercontent.com/91017318/226846170-5532920c-050a-4839-8c1d-dcd327b8db57.png">

**Comment by nlohmann** [comment, marked as answer]
You try to read from stream `f` twice. Try calling

```cpp
f.clear();
f.seekg(0);
```

before calling `parse`.

**Reply by MikailKahya** [reply]
This indeed fix it, ty. I do wonder why this would be the solution to this issue

**Reply by gregmarr** [reply]
That's how streams work, they remember what you've read.  To read the same thing more than once requires that you reset the stream to the beginning between reads.  Your other options are to read the stream into a string and then call `accept()` and `parse()` on the stream, or just call `parse()` and handle any exceptions.
